content/python/firewalls.md

```diff
@@ -0,0 +1,454 @@
+            # It is always necessary to delete this key to perform the firewall update request
+            del rule['status']
+
+        rules_template = {
+            'bypassFlag': False,
+            'networkComponentFirewallId': id_firewall,
+            'rules': rules
+        }
+        new_rules = self.client.call('SoftLayer_Network_Firewall_Update_Request', 'createObject', rules_template)
+        self.env.fout(formatting.iter_to_table(new_rules))
+
+```
+## Bypass Rules for a Firewall in a Bypass
+Bypass rules for a firewall in a VSI.
+### Python
+```python
+
+    def bypass_rules_for_a_firewall_in_a_bypass(self, id_firewall):
+        rules = self.client.call('SoftLayer_Network_Component_Firewall', 'getRules', id=id_firewall)
+
+        for rule in rules:
+            # It is always necessary to delete this key to perform the firewall update request
+            del rule['status']
+
+        rules_template = {
+            'bypassFlag': True,
+            'networkComponentFirewallId': id_firewall,
+            'rules': rules
+        }
+        new_rules = self.client.call('SoftLayer_Network_Firewall_Update_Request', 'createObject', rules_template)
+        self.env.fout(formatting.iter_to_table(new_rules))
+
+```
+
+## Get Firewall Rules for a Vsi
+Get the associated Firewall's rules for a vsi.
+### Python
+```python
+
+    def get_rules_for_a_vsi(self, primary_ip_address):
+        vsi = self.client.call('SoftLayer_Virtual_Guest', 'findByIpAddress', primary_ip_address)
+        firewall = self.client.call('SoftLayer_Virtual_Guest', 'getFirewallServiceComponent', id=vsi['id'])
+        rules = self.client.call('SoftLayer_Network_Component_Firewall', 'getRules', id=firewall['id'])
+        self.env.fout(formatting.iter_to_table(rules))
+
+```
+
+## Get Firewall Logs
+Get the firewall logs for an arbitrary IP address.
+### Python
+```python
+
+    def get_firewall_logs(self, primary_ip_address):
+        subnetIp = self.client.call('SoftLayer_Network_Subnet_IpAddress', 'getByIpAddress', primary_ip_address)
+        self.env.fout(formatting.iter_to_table(subnetIp))
+        logs = self.client.call('SoftLayer_Network_Subnet_IpAddress', 'getSyslogEventsOneDay', id=subnetIp['id'], limit=10)
+        self.env.fout(formatting.iter_to_table(logs))
+
+```
+
+## Get Firewall Ip Address
+Get the IP address from a VLan.
+
+The script lists the IP address of VLAN, it makes a single call to the `SoftLayer_Network_Vlan::getFirewallProtectableIpAddresses` method.
+
+### Python
+```python
+
+    def get_firewall_ip_address(self, primary_ip_address):
+        vlan = self.client.call('SoftLayer_Network_Vlan', 'getVlanForIpAddress', primary_ip_address)
+        self.env.fout(formatting.iter_to_table(vlan))
+        firewall = self.client.call('SoftLayer_Network_Vlan', 'getFirewallProtectableIpAddresses', id=vlan['id'], limit=10)
+        self.env.fout(formatting.iter_to_table(firewall))
+
+```
+
+## Edit Standard Rules
+Edit Standard Rule`.
```

Comment:
Remove the single quote before the end point, after that it looks good to me

Reply:
The single quote was removed